src/kernel/messages.c

```
@@ -194,13 +194,15 @@ message *msg_message(const char *name, const char *sig, ...)
         }
         else {
             log_error("invalid parameter %s for message type %s\n", paramname, mtype->name);
+            assert(!"invalid parameter for message type");
```

Ja, das kann ich nachvollziehen. Leider führte die Version ohne den Assert bei mir zu einer ziemlich langen Debug-Sitzung, bis ich herausgefunden hatte, dass ich (AFAIC) eine message mit falschen Parametern erzeugt hatte. Mit dem assert hätte ich das gleich gemerkt ("fail fast"). Ins Log habe ich natürlich nicht geschaut (falls es das im Test überhaupt gibt).

Natürlich ist das hier _nicht direkt_ ein Programmierfehler, sondern der Fehler kommt erst durch Zusammenspiel von xml-Konfiguration und Code.

Fazit: Ich hänge nicht an dem assert, aber es hätte mir das Leben leichter gemacht und eine halbe Stunde Lebenszeit gebracht...